File: watch/js/entity.js

function VisaulizeCircle(x, y, r, fillStyle, strokeStyle) {
  context.beginPath();
  context.arc(x, y, r, 0, 2 * Math.PI);

  context.fillStyle = fillStyle;
  context.fill();

  context.strokeStyle = strokeStyle;
  context.stroke();
}

var Entity  = function() {
  this.fillStyle = 'black';
	this.strokeStyle = 'black';
  this.orbitFillStyle = 'transparent';
	this.orbitStrokeStyle = 'silver';

	this.angle = 0.0;
	this.angleSpeed = 0.0;
	this.orbit = 0.0;
	this.orbitSpeed = 0.0;

	this.radius = 0.0;

	this.next = null;

	this.Update = function(timeEllapsed) {
		this.angle += this.angleSpeed * timeEllapsed;
		this.orbit += this.orbitSpeed * timeEllapsed;
		this.orbit = Math.max(this.orbit, 0.0);
	}

	this.Draw = function(scale) {
		var x = scale * this.GetX();
		var y = scale * this.GetY();
		var r = scale * this.radius;
    VisaulizeCircle(x, y, r, this.fillStyle, this.strokeStyle);
	}

	this.DrawOrbit = function(scale) {
    var x = scale * 0.0;
		var y = scale * 0.0;
		var r = scale * this.orbit;
    VisaulizeCircle(x, y, r, this.orbitFillStyle, this.orbitStrokeStyle);
	}

	this.GetX = function() {
		return this.orbit * Math.cos(this.angle);
	}

	this.GetY = function() {
		return this.orbit * Math.sin(this.angle);
	}

	this.GetDistance = function(entity) {
		var x = this.GetX() - entity.GetX();
		var y = this.GetY() - entity.GetY();
		return Math.sqrt(x * x + y * y);
	}
}

File: MoveEdge/Entity.cs
using System;
using System.Drawing;

namespace Prototype
{
    public class Entity
    {
        public float r = 10;
        public float x_, _x, y_, _y;
        public float vx_, _vx, vy_, _vy;

        public void Init(float x, float y, float a = 0)
        {
            x_ = x - r * (float)Math.Cos(a);
            _x = x + r * (float)Math.Cos(a);
            y_ = y - r * (float)Math.Sin(a);
            _y = y + r * (float)Math.Sin(a);
        }

        public void Move()
        {
            float a = (float)Math.Atan2(_y - y_, _x - x_);

            x_ += vx_ * (float)Math.Cos(a) - vy_ * (float)Math.Sin(a);
            y_ += vx_ * (float)Math.Sin(a) + vy_ * (float)Math.Cos(a);

            _x += _vx * (float)Math.Cos(a) - _vy * (float)Math.Sin(a);
            _y += _vx * (float)Math.Sin(a) + _vy * (float)Math.Cos(a);

            float px = 0.5f * (x_ + _x);
            float py = 0.5f * (y_ + _y);

            float vx = _x - x_;
            float vy = _y - y_;
            float k = r / (float)Math.Sqrt(vx * vx + vy * vy);
            vx *= k; vy *= k;

            x_ = px - vx; y_ = py - vy;
            _x = px + vx; _y = py + vy;
        }

        public void Draw(Graphics g, Pen pen)
        {
            g.DrawLine(pen, x_, y_, _x, _y);
        }
    }
}
